.pre-commit-config.yaml

```
@@ -0,0 +1,36 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+        - id: trailing-whitespace
+        - id: check-executables-have-shebangs
+        - id: check-json
+        - id: check-case-conflict
+        - id: check-toml
+        - id: check-merge-conflict
+        - id: check-xml
+        - id: check-yaml
+        - id: end-of-file-fixer
+        - id: check-symlinks
+        - id: fix-encoding-pragma
+        - id: mixed-line-ending
+        - id: pretty-format-json
+          args: [--autofix]
+  - repo: https://gitlab.com/pycqa/flake8
+    rev: 3.9.2
+    hooks:
+        - id: flake8
+          args: [
+              '--max-line-length=150',
+              '--ignore=E203,E402,E501,E800,W503,W391,E261',
+              '--select=B,C,E,F,W,T4,B9'
+              ]
```

FWIW these can be specified as:
```suggestion
          args:
            - --max-line-length=150
            - --ignore=E203,E402,E501,E800,W503,W391,E261
            - --select=B,C,E,F,W,T4,B9
```
(same with the `black` args below)